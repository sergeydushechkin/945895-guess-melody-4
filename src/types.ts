export enum GameType {
  ARTIST = `artist`,
  GENRE = `genre`,
}

export interface AnswerArtist {
  artist: string,
  picture: string,
}

export interface AnswerGenre {
  genre: string,
  src: string,
}

export interface QuestionArtist {
  answers: AnswerArtist[],
  song: {
    artist: string,
    src: string,
  },
  type: GameType.ARTIST,
}

export interface QuestionGenre {
  answers: AnswerGenre[],
  genre: string,
  type: GameType.GENRE,
}
