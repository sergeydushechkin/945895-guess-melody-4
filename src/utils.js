export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const noop = () => {
  // do nothing
};
