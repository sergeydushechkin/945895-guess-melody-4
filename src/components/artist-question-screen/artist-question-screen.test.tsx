import * as React from "react";
import * as renderer from "react-test-renderer";
import ArtistQuestionScreen from "./artist-question-screen";
import {GameType, QuestionArtist} from "../../types";
import {noop} from "../../utils";

const question: QuestionArtist = {
  type: GameType.ARTIST,
  song: {
    artist: `Jim Beam`,
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  },
  answers: [{
    picture: `https://api.adorable.io/avatars/128/0`,
    artist: `John Snow`,
  }, {
    picture: `https://api.adorable.io/avatars/128/1`,
    artist: `Jack Daniels`,
  }, {
    picture: `https://api.adorable.io/avatars/128/2`,
    artist: `Jim Beam`,
  }],
};

it(`ArtistQuestionScreen is rendered correctly`, () => {
  const tree = renderer.create(
      <ArtistQuestionScreen
        question={question}
        onAnswer={noop}
        renderPlayer={() => null}
      />, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
