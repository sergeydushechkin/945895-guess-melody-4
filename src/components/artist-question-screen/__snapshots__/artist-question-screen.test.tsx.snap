// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen is rendered correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    />
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="answer-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="John Snow"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/0"
        />
        John Snow
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="answer-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Jack Daniels"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/1"
        />
        Jack Daniels
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="answer-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Jim Beam"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/2"
        />
        Jim Beam
      </label>
    </div>
  </form>
</section>
`;
